src/main/connect/connectionManager.ts

```diff
@@ -257,47 +374,108 @@ export class ConnectionManager implements ExtensionComponent {
         await this._context.globalState.update(ConnectionManager.RT_URL_KEY, this._rtUrl);
     }
 
-    private async setUsername(prompt: boolean): Promise<boolean> {
-        if (prompt) {
-            this._username =
-                (await vscode.window.showInputBox({
-                    prompt: 'Enter username',
-                    value: this._username,
-                    ignoreFocusOut: true,
-                    validateInput: ConnectionUtils.validateFieldNotEmpty
-                })) || '';
-        } else {
-            this._username = (await this._context.globalState.get(ConnectionManager.XRAY_USERNAME_KEY)) || '';
-        }
+    private async setUsernameFromFilesystem(): Promise<boolean> {
+        this._username = (await this._context.globalState.get(ConnectionManager.XRAY_USERNAME_KEY)) || '';
+        return !!this._username;
+    }
+
+    private async promptUsername(): Promise<boolean> {
+        this._username =
+            (await vscode.window.showInputBox({
+                prompt: 'Enter username',
+                value: this._username,
+                ignoreFocusOut: true,
+                validateInput: ConnectionUtils.validateFieldNotEmpty
+            })) || '';
         return !!this._username;
     }
 
     private async storeUsername() {
         await this._context.globalState.update(ConnectionManager.XRAY_USERNAME_KEY, this._username);
     }
 
-    private async setPassword(prompt: boolean): Promise<boolean> {
-        if (prompt) {
-            this._password =
-                (await vscode.window.showInputBox({
-                    prompt: 'Enter password',
-                    password: true,
-                    ignoreFocusOut: true,
-                    validateInput: ConnectionUtils.validateFieldNotEmpty
-                })) || '';
-        } else {
-            this._password =
-                (await keytar.getPassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._url, this._username))) ||
-                (await keytar.getPassword(ConnectionManager.SERVICE_ID, this.createAccountId(this._xrayUrl, this._username))) ||
-                '';
+    private async setPasswordFromFilesystem(): Promise<boolean> {
+        this._password = await this.getSecretFromFilesystem(this._username);
+        return !!this._password;
+    }
+
+    private async setAccessTokenFromFilesystem(): Promise<boolean> {
+        this._accessToken = await this.getSecretFromFilesystem(ConnectionManager.ACCESS_TOKEN_FS_KEY);
+        return !!this._accessToken;
+    }
+
+    // Password and access token are saved in keychain with an account that is a hash of url and another string.
+    // For password - username, access token - a constant.
+    private async getSecretFromFilesystem(keyPair: string): Promise<string> {
```

Comment:
(fill in the missing fields)
```suggestion
    /**
    * Password and access token are saved in KeyStore with an account that is a hash of url and another string.
    * For password - username, access token - a constant.
    * @param - keyPair ...
    * @returns ...
    */
    private async getSecretFromKeyStore(keyPair: string): Promise<string> {
```